Sources/MongoSwift/Operations/ListCollectionsOperation.swift

```diff
@@ -117,20 +117,20 @@ internal struct ListCollectionsOperation: Operation {
             }
         }
 
-        let initializer = { (conn: Connection) -> OpaquePointer in
-            self.database.withMongocDatabase(from: conn) { dbPtr in
-                guard let collections = mongoc_database_find_collections_with_opts(dbPtr, opts._bson) else {
-                    fatalError(failedToRetrieveCursorMessage)
-                }
-                return collections
+        let indexes: OpaquePointer = self.database.withMongocDatabase(from: connection) { dbPtr in
+            guard let collections = mongoc_database_find_collections_with_opts(dbPtr, opts._bson) else {
+                fatalError(failedToRetrieveCursorMessage)
             }
+            return collections
         }
+
         if self.nameOnly {
             let cursor: MongoCursor<Document> = try MongoCursor(
+                stealing: indexes,
+                connection: connection,
                 client: self.database._client,
                 decoder: self.database.decoder,
-                session: session,
-                initializer: initializer
+                session: session
             )
             return try .names(cursor.map {
```

Comment:
in future PR I will update this so we aren't iterating the cursor and generating new operations/tieing up an extra thread from within this operation. and also update it so we are making sure this cursor didn't encounter an error while iterating.